src/mod_ping.erl

```diff
@@ -120,6 +122,19 @@ iq_ping(Acc, _From, _To, #iq{sub_el = SubEl} = IQ, _) ->
 %% Hook callbacks
 %%====================================================================
 
+-spec filter_local_packet(Acc, Params, Extra) -> {ok, Acc} | {stop, drop} when
+      Acc :: mongoose_hooks:filter_packet_acc(),
+      Params :: map(),
+      Extra :: gen_hook:extra().
+filter_local_packet({_, _, _, Stanza} = Acc, _Params, _Extra) ->
+    case is_ping_error(Stanza) of
```

Comment:
Instead of this, I think we need to call the logic that is currently in `user_send_iq/3`, because when the user sends an error stanza, it makes sense to cancel the timeout. We might then even disconnect the session if the `kill` action is configured.

Reply:
@jacekwegr, is this comment addressed?

Reply:
For me it is addressed, because the user-sent error is handled in `handle_ping_response`, line 151.